### DISCUSSION TITLE
renderView1.InteractionMode = '2D'

### DISCUSSION THREAD

**Discussion by FrankFrank9** [pyvista/pyvista#5322] [Q&A] [answered]
Hello,

what is the pyvista equivalent of a traced state command in paraview `renderView1.InteractionMode = '2D'`?

I have not been able to find anything suitable

Best regards

**Comment by MatthewFlamm** [comment]
I would also like to have this feature.  When I looked into this before, I found that this is a Paraview specific interaction mode that isn't defined in vtk.  I _think_ we may have to define a custom interactor style that mimics the Paraview 2D style.

Part of the Paraview style is to use parallel projection, which you do on the Plotter object directly  `enable_parallel_projection`.  Another part is to change the mouse bindings, which would have to be custom style.  Related to those mouse bindings is that the view direction is pinned during rotation.

One related interaction style in pyvista is the [Rubber Band 2D style](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.enable_rubber_band_2d_style.html), but I find it not fully satisfying.  There is no rotate and the panning works somewhat but is confusing when not looking in a Cartesian aligned direction, e.g. +x.  The bindings are different to the Paraview style, so there is a mental barrier in using it effectively too.

**Reply by MatthewFlamm** [reply]
I've tested https://vtk.org/doc/nightly/html/classvtkParallelCoordinatesInteractorStyle.html#details by adding it as a style in pyvista and it doesn't work for me like the documentation linked.  I suspect this is a pyvista issue.

**Comment by PBrockmann** [comment]
Any news on this issue ?

From the documentation on geovista
```
plotter = gv.GeoPlotter(crs="+proj=robin lon_0=-90")
plotter.add_mesh(sea_region, show_edges=True)
plotter.add_coastlines()
plotter.add_base_layer(texture=gv.natural_earth_hypsometric())
plotter.view_xy()
plotter.show_axes()
plotter.show()
```

Ok but it is still in 3D.
With vtk, you can specify an interactor 
```
imagestyle = vtk.vtkInteractorStyleImage()
inter.SetInteractorStyle(imagestyle)
```

How to do this with pyvista/geovista ?

**Reply by MatthewFlamm** [reply, marked as answer]
This is implemented in https://github.com/pyvista/pyvista/pull/5336, and will be in the next minor release 0.44.0

**Reply by PBrockmann** [reply]
Yes ! 
I was testing some code from https://github.com/pyvista/pyvista/pull/5336/commits
```
pl.enable_parallel_projection()
pl.enable_2d_style()
```
will work ?
Many thanks anyway

**Reply by MatthewFlamm** [reply]
It should but you need to install a version of PyVista from the `main` branch rather than from pypi, etc.

**Reply by PBrockmann** [reply]
Tested with `pip install pyvista==0.44.dev0`
It works nicelly  !

Great !

**Reply by MatthewFlamm** [reply]
> `pip install pyvista==0.44.dev0`

I didn't even know we had this.  Glad it works!

**Reply by MatthewFlamm** [reply]
You can check the dev documentation at https://docs.pyvista.org/version/dev/api/plotting/_autosummary/pyvista.Plotter.enable_2d_style.html#pyvista.Plotter.enable_2d_style .

Anyone coming to this in the future, please check in the stable version of the documentation.

**Comment by tkoyama010** [comment]
> > `pip install pyvista==0.44.dev0`
> 
> I didn't even know we had this. Glad it works!

Yes. It's an awesome feature of PyPI to recognize main branch as pre-release :)
